src/views/Space.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <Layout>
+    <template #sidebar-left>
+      <BlockSpace :space="space" />
+    </template>
+    <template #content-right>
+      <div class="px-4 px-md-0 mb-3 d-flex">
+        <div class="flex-auto">
+          <div v-text="space.name" />
+          <div class="d-flex flex-items-center flex-auto">
+            <h2>{{ $t('proposals.header') }}</h2>
+          </div>
+        </div>
+        <UiDropdown
+          top="3.5rem"
+          right="1.25rem"
+          @select="selectState"
+          :items="[
+            { text: $t('proposals.states.all'), action: 'all' },
+            { text: $t('proposals.states.active'), action: 'active' },
+            { text: $t('proposals.states.pending'), action: 'pending' },
+            { text: $t('proposals.states.closed'), action: 'closed' }
+          ]"
+        >
+          <UiButton class="pr-3">
+            {{ $t(`proposals.states.${filterBy}`) }}
+            <Icon size="14" name="arrow-down" class="mt-1 mr-1" />
+          </UiButton>
+        </UiDropdown>
+      </div>
+
+      <Block v-if="loading" :slim="true">
+        <RowLoading class="my-2" />
+      </Block>
+
+      <NoResults :block="true" v-else-if="proposals.length < 1" />
+      <div v-else>
+        <Block :slim="true" v-for="(proposal, i) in proposals" :key="i">
+          <TimelineProposal :proposal="proposal" />
+        </Block>
+      </div>
+      <div
+        style="height: 10px; width: 10px; position: absolute"
+        id="endofpage"
+      />
+      <Block v-if="loadingMore && !loading" :slim="true">
+        <RowLoading class="my-2" />
+      </Block>
+    </template>
+  </Layout>
+</template>
+
+<script>
+import { computed, onMounted, ref } from 'vue';
+import { useRoute } from 'vue-router';
+import { useInfiniteLoader } from '@/composables/useInfiniteLoader';
+import { subgraphRequest } from '@snapshot-labs/snapshot.js/src/utils';
+import { useScrollMonitor } from '@/composables/useScrollMonitor';
+import { useStore } from 'vuex';
+
+// Persistent filter state
+const filterBy = ref('all');
+
+export default {
+  setup() {
+    const store = useStore();
+    const route = useRoute();
+
+    const loading = ref(false);
+    const proposals = ref([]);
+    const space = computed(() => store.state.app.spaces[route.params.key]);
+
+    // Infinite scroll with pagination
+    const {
+      loadBy,
+      limit,
+      loadingMore,
+      stopLoadingMore,
+      loadMore
+    } = useInfiniteLoader();
+
+    useScrollMonitor(() =>
```

Comment:
@bonustrack if there aren't any major changes you are looking to make in this PR I will start a PR where a query with Apollo and move this into a composable that can be used on timeline also.

Reply:
I will merge this PR first then you can do that change with composable